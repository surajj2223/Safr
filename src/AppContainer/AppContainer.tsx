import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from 'styled-components';

import themes from '../util/theme';
import Navigation from '../Navigation/Navigation';

export default () => {
  const colorScheme = useColorScheme();
  const theme = colorScheme
    ? colorScheme === 'dark'
      ? themes.dark
      : themes.light
    : themes.light;

  return (
    <ThemeProvider theme={theme}>
      <Navigation />
    </ThemeProvider>
  );
};
