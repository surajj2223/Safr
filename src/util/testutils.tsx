import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import themes, { Theme } from './theme';

interface ThemeWrapperProps {
  children: React.ReactNode;
  theme?: Theme;
}

export const ThemeWrapper: React.FunctionComponent<ThemeWrapperProps> = ({
  children,
  theme = themes.light,
}: ThemeWrapperProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
