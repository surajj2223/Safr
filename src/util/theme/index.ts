import light from './light';
import dark from './dark';

export interface Theme {
  primaryColor: string;
  ascentColor: string;
}

export default {
  light,
  dark,
};
