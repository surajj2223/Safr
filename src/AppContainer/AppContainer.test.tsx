import React from 'react';
import { render } from '@testing-library/react-native';
import AppContainer from './AppContainer';
import { ThemeWrapper } from '../util/testutils';

describe('App Container', () => {
  test('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeWrapper>
        <AppContainer />
      </ThemeWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
