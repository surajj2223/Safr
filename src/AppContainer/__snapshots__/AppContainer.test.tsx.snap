// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container should match snapshot with light theme 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#67AC90",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 16,
        },
      ]
    }
  >
    Hi there!
  </Text>
</RCTSafeAreaView>
`;
